M2/Macaulay2/m2/mutablemat.m2

```diff
@@ -16,6 +16,9 @@ precision MutableMatrix := precision @@ ring
 expression MutableMatrix := m -> MatrixExpression append(applyTable(entries m, expression), symbol MutableMatrix => true)
 texMath MutableMatrix := m -> texMath expression m
 net MutableMatrix := m -> net expression m
+describe MutableMatrix := m -> Describe FunctionApplication(
+    mutableMatrix, describe matrix m)
+toExternalString MutableMatrix := toString @@ describe
```

Comment:
I think I'd rather have a method that checks whether a matrix is dense or sparse in memory, and if it's sparse describe and toExternalMatrix should do something different. Do you need this change now, or are you okay with keeping this open until we can expand it?

Reply:
I'm in no rush -- I just noticed that the current behavior raised an error!

Reply:
Well, now that you point it out, maybe it _should_ raise an error, just a different error:
```m2
i1 : toExternalString new MutableList from {}
stdio:1:1:(3): error: anonymous mutable list cannot be converted to external string

i2 : toExternalString new MutableHashTable from {}
stdio:2:1:(3): error: anonymous mutable hash table cannot be converted to external string

i3 : toExternalString new MutableMatrix from id_(ZZ^2)
stdio:3:1:(3): error: can't convert local symbol or invisible global symbol 'RawMutableMatrix' to external string
null: here is the first use of 'RawMutableMatrix'

```

Reply:
What's the rationale for the mutable list/hash table errors?

Reply:
I think the point of `toExternalString` is that it should satisfy `X === value toExternalString X`, which can never be the case for mutable objects.

Reply:
Ok, that makes sense

Reply:
Actually what I said can't be true because of these:
```m2
i16 : select(methods toExternalString, m -> ancestor(MutableHashTable, m#1))

o16 = {0 => (toExternalString, FractionField)   }
      {1 => (toExternalString, GaloisField)     }
      {2 => (toExternalString, Monoid)          }
      {4 => (toExternalString, PolynomialRing)  }
      {5 => (toExternalString, QuotientRing)    }
      {7 => (toExternalString, Variety)         }
```
But I think these are sort of the exceptions?